custom/icds_reports/utils/__init__.py

```diff
@@ -837,13 +837,6 @@ def track_time(func):
     """A decorator to track the duration an aggregation script takes to execute"""
     from custom.icds_reports.models import AggregateSQLProfile
 
-    def get_async_indicator_time():
-        try:
-            return AsyncIndicator.objects.exclude(date_queued__isnull=True)\
-                .aggregate(Max('date_created'))['date_created__max'] or datetime.now()
```

Comment:
I'm guessing this query is the part that took a long time. Do we know why it took so long?

Reply:
It looked like locking issues. There were a sizable number of queries waiting on `LWLockTranche`. I am not sure why these queries lock as it looks like a simple select, but it may have to do with other inserts happening at the same time. I am honestly quite confused by that, but the queries had been waiting for significant periods, and locking issues also explains why it appears to all clear at once. It tends to resolve in waves, which would make sense if all were waiting on the same lock.